### PR TITLE
feat:Rotate image

### PR DESCRIPTION
**Pull request by DanieliusAsm**
### What
- Rotate images by 90, 180 and 270 degrees

How it works:

- Gets the subdomain lc. for example ru-en.openfoodfacts will return ru.
- Sends request to get product images.
- Checks if theres a front_ru available, otherwise use whatever is available first like front_en
- Gets the imgid to use for rotation from front_lc
- Rotates by sending JSON request
- Also rotates image on client side

### Screenshot
image

Part of https://github.com/openfoodfacts/openfoodfacts-server/issues/8135

### CODE REVIEW

**Comment by DanieliusAsm** [comment]
Im having some issues:

- From looking at [api docs](https://openfoodfacts.github.io/openfoodfacts-server/api/ref-v2/#get-/cgi/product_image_crop.pl ) I see that id like (front_en) and imageid like '1' is required.
- I want to rotate the currently selected image, I dont know its id.
- I also want to rotate front_lc. I dont know which lc is used.
- I can try rotating front_en by default but that causes problems if you use a different than world.openfoodfacs subdomain like it.openfoodfacts

**Comment by DanieliusAsm** [comment]
I am having some troubles:

- Rotate image needs id and imageid. I Need to rotate the currently selected image.
- the id might be front_it but I try and rotate front_en

File: OpenFoodFactsPower.user.js [hunk 1]
```diff
@@ -1977,12 +2001,87 @@ ul#products_match_all > li > a > span { display: table-cell; width:   70%;  vert
...
     }
 
+    //shows HungerGames logo, rotate buttons
+    function showListButtons(){
+        let languageCode = getSubdomainLanguageCode();
+
+        $("ul[id^='products_'].search_results li[data-code]").each(function(index, element) {
+            let barcode = $(this).attr('data-code');
+            $(this).append('<a class="list_hunger_games_logo_search" alt="Hunger games logo search" title="Hunger games logo search" href="https://hunger.openfoodfacts.org/logos/search?barcode='+barcode+'"><span class="material-icons">image_search</span></a>');
+            $(this).append('<a class="list_rotate_image_270" alt="Rotate at 270°" title="Rotate at 270°"><span class="material-icons" style="transform: rotate(-90deg);">rotate_left</span></a>');
```
Comment (CharlesNepote) [inline OpenFoodFactsPower.user.js:2016]: I feel the icons you're using are hard to read/understand. I suggest these other icons, feel free to adopt or reject it:
```suggestion
            $(this).append('<a class="list_rotate_image_270" alt="Rotate at -90°" title="Rotate at -90°"><span class="material-icons"  style="transform: scaleX(-1);">redo</span></a>');
```

File: OpenFoodFactsPower.user.js [hunk 2]
```diff
@@ -1977,12 +2001,87 @@ ul#products_match_all > li > a > span { display: table-cell; width:   70%;  vert
...
+            $(this).append('<a class="list_rotate_image_180" alt="Rotate at 180°" title="Rotate at 180°"><span class="material-icons">rotate_left</span></a>');
```
Comment (CharlesNepote) [inline OpenFoodFactsPower.user.js:2017]:
```suggestion
            $(this).append('<a class="list_rotate_image_180" alt="Rotate at 180°" title="Rotate at 180°"><span class="material-icons">rotate_right</span></a>');
```

File: OpenFoodFactsPower.user.js [hunk 3]
```diff
@@ -1977,12 +2001,87 @@ ul#products_match_all > li > a > span { display: table-cell; width:   70%;  vert
...
+            $(this).append('<a class="list_rotate_image_90" alt="Rotate at 90°" title="Rotate at 90°"><span class="material-icons" style="transform: rotate(90deg);">rotate_right</span></a>');
```
Comment (CharlesNepote) [inline OpenFoodFactsPower.user.js:2018]:
```suggestion
            $(this).append('<a class="list_rotate_image_90" alt="Rotate at 90°" title="Rotate at 90°"><span class="material-icons">redo</span></a>');
```

Comment (CharlesNepote) [reply OpenFoodFactsPower.user.js:2018]: Here is the result:
image

**Review by CharlesNepote — CHANGES_REQUESTED** [review]
This is really a great job!! I have suggested to change the icons but please do what you prefer.

**Comment by DanieliusAsm** [comment]
@CharlesNepote your icons should be easier to understand since the ones I am using show a 180 degree turn. Also I probably could reduce it to 2 rotate icons. 

When you click to rotate at first it sends request to get images, from there I can get the current rotation and end up using 2 instead of 3 buttons. Just need to try it out and see if I can get it working.

**Pull request merged**